### PR TITLE
Validate `Wallet` and `Membership` values

### LINKED ISSUE #87: Validate `Wallet` and `Membership` values on first fetch
Validate `Wallet` and `Membership` values (not just their existence on the blockchain) on first fetch, that is when `validated` is `false`

### PR DESCRIPTION
**Pull request by NikhilSharma03**
## Description

- This PR resolves #87

## Validating
- [x] `name`
- [x] `totalShares`
- [x] `memberShipType`
- [x] `SPLToken`

### CODE REVIEW

**Comment by NikhilSharma03** [comment]
@abdelhakeem Need to Fetch SPL Token Details (if wallet accept). Any method which could help to fetch the detail?

**Comment by abdelhakeem** [comment]
> @abdelhakeem Need to Fetch SPL Token Details (if wallet accept). Any method which could help to fetch the detail?

Well, I don't think we can directly obtain the SPL token(s) that the wallet accepts unless we use [`getProgramAccounts`](https://solana-labs.github.io/solana-web3.js/classes/Connection.html#getProgramAccounts) to fetch all the related [`FanoutMint`](https://github.com/GlassEaters/hydra/blob/main/packages/sdk/src/generated/accounts/FanoutMint.ts) accounts. But we can verify that the SPL token in the database is actually accepted by the wallet by trying to fetch the corresponding `FanoutMint` object:

```typescript
const [fanoutMintKey] = FanoutClient.fanoutForMintKey(fanoutKey, splTokenKey)
const fanoutMint = FanoutMint.fromAccountAddress(connection, fanoutMintKey)
```

**Comment by NikhilSharma03** [comment]
@abdelhakeem Added validation for SPL Token. Could you please review it?

File: pages/api/wallets/[viewWalletPubkey].ts [hunk 1]
```diff
@@ -79,17 +89,38 @@ export default async function handler(
...
-          // TODO
-
+          // Fetch SPL Details and Update wallet if it accept SPL token
+          if (element.acceptSPL) {
+            let splToken = await fetchSPLDetails(
+              cluster,
+              viewWalletPubkey,
+              element.splToken
+            )
+            if (splToken) {
```
Comment (abdelhakeem) [inline pages/api/wallets/[viewWalletPubkey].ts:99]: Looks perfect! One more thing, if the SPL token in the database is not accepted by the wallet (i.e.; `splToken` here is `null`), we should remove the SPL token from the database and set `acceptSPL` to false.

Comment (NikhilSharma03) [reply pages/api/wallets/[viewWalletPubkey].ts:99]: ✅

**Pull request merged**